src/plans/create/CreatePlanWizard.tsx

```diff
@@ -9,44 +10,92 @@ import { GeneralFormFieldId } from './steps/general-information/constants';
 import GeneralInformationStep from './steps/general-information/GeneralInformationStep';
 import HooksStep from './steps/hooks/HooksStep';
 import MigrationTypeStep from './steps/migration-type/MigrationTypeStep';
+import { NetworkMapFieldId } from './steps/network-map/constants';
 import NetworkMapStep from './steps/network-map/NetworkMapStep';
 import OtherSettingsStep from './steps/other-settings/OtherSettingsStep';
+import ReviewStep from './steps/review/ReviewStep';
+import { StorageMapFieldId } from './steps/storage-map/constants';
 import StorageMapStep from './steps/storage-map/StorageMapStep';
+import { VmFormFieldId } from './steps/virtual-machines/constants';
 import VirtualMachinesStep from './steps/virtual-machines/VirtualMachinesStep';
 import { firstStep, planStepNames, planStepOrder, PlanWizardStepId } from './constants';
 import CreatePlanWizardFooter from './CreatePlanWizardFooter';
 import { useCreatePlanForm, useDefaultFormValues } from './hooks';
-import { hasWarmMigrationProviderType } from './utils';
+import { getCreatedPlanPath, handlePlanSubmission, hasWarmMigrationProviderType } from './utils';
 
 import './CreatePlanWizard.style.scss';
 
 const CreatePlanWizard: FC = () => {
   const { t } = useForkliftTranslation();
+  const history = useHistory();
+  const [currentStep, setCurrentStep] = useState<WizardStepType>(firstStep);
+  const [createPlanError, setCreatePlanError] = useState<Error>();
+  const [isCreating, setIsCreating] = useState(false);
+
   const defaultValues = useDefaultFormValues();
   const form = useCreatePlanForm({
     defaultValues,
     mode: 'onChange',
   });
-  const [currentStep, setCurrentStep] = useState<WizardStepType>(firstStep);
-  const { control, formState, watch } = form;
-  const formValues = watch();
-  const sourceProvider = useWatch({ control, name: GeneralFormFieldId.SourceProvider });
+  const hasCreatePlanError = Boolean(createPlanError?.message);
+
+  const { control, formState } = form;
+  const [planName, planProject, sourceProvider, targetProvider, vms, networkMap, storageMap] =
+    useWatch({
+      control,
+      name: [
+        GeneralFormFieldId.PlanName,
+        GeneralFormFieldId.PlanProject,
+        GeneralFormFieldId.SourceProvider,
+        GeneralFormFieldId.TargetProvider,
+        VmFormFieldId.Vms,
+        NetworkMapFieldId.NetworkMap,
+        StorageMapFieldId.StorageMap,
```

Comment:
if its all the values, u need to declare it by name?
maybe just take them all and send to handlesubmit as 1 arg.

Reply:
I think I can get it to work by using `getValues` returned from the `form` object on line 42 and pass that in, but there will still be a watch here, since `planName, planProject, sourceProvider`, these 3 are used in this file outside of that util.